### PR TITLE
Add new IPAM data to etcd structure doc

### LINKED ISSUE #671: etcdStructure.md is out of date
The structure of etcd has changed since this doc was written.  The ipam paths are significantly different with the introduction of ipam blocks, and some naming has also changed, such as changing `/calico/v1/host/<host>/bird_ip` to `/calico/v1/host/<host>/bird_ip`

### PR DESCRIPTION
**Pull request by robbrockbank**
Fixes #671

### CODE REVIEW

File: docs/etcdStructure.md [hunk 1]
```diff
@@ -9,148 +9,233 @@
...
+       |  |     `--block
+       |  |         `--<CIDR>  # One per block, key is CIDR with '/' replaced
+       |  |                    # by '-', value is JSON Allocation Block object (see below)
+       |  |--handle
+       |  |  `--<Handle ID>    # One per handle, value is JSON Allocation Handle
+       |  |                    # object (see below)
+       |  |--host
+       |  |  `--<hostname>
+       |  |     |--ipv4
+       |  |     |  `--<CIDR>   # CIDR matching the Allocation Block with this host
```
Comment (TrimBiggs) [inline docs/etcdStructure.md]: This and `ipv6` CIDRs are both one layer deeper under `block:
`/calico/ipam/v2/host/7fd0dabeeb24/ipv4/block/192.168.1.0-26`

File: docs/etcdStructure.md [hunk 2]
```diff
@@ -9,148 +9,233 @@
...
-	            |  `--<BGP peer IPv4 address>  # JSON BGP peer configuration (see below)
-	            `--peer_v6  # Host specific IPv6 BGP peers
-	               `--<BGP peer IPv6 address>  # JSON BGP peer configuration (see below)
+    
+    +--calico  # root namespace
+       |
+       |--v1
+       |  |--config
+       |  |  |--InterfacePrefix # the prefix for Calico interface names
+       |  |  `--LogSeverityFile # Log severity level for writing to file e.g. "DEBUG"
```
Comment (TrimBiggs) [inline docs/etcdStructure.md]: There are a few more of these now:
```
/calico/v1/config/LogFilePath
/calico/v1/config/IpInIpEnabled
/calico/v1/config/LogSeverityScreen
```

File: docs/etcdStructure.md [hunk 3]
```diff
@@ -9,148 +9,233 @@
...
+       |  |--host
+       |  |  `--<hostname>      # one for each Docker host in the cluster
+       |  |     |--config       # Host level config
+       |  |     |  `--marker
+       |  |     |--ipv4_address # the next hop IP address of this host
```
Comment (TrimBiggs) [inline docs/etcdStructure.md]: I think this should be called `bird_ip`:
`/calico/v1/host/7fd0dabeeb24/bird_ip`

File: docs/etcdStructure.md [hunk 4]
```diff
@@ -9,148 +9,233 @@
...
+    +--calico  # root namespace
+       |
+       |--v1
+       |  |--config
+       |  |  |--InterfacePrefix # the prefix for Calico interface names
+       |  |  `--LogSeverityFile # Log severity level for writing to file e.g. "DEBUG"
+       |  |--host
+       |  |  `--<hostname>      # one for each Docker host in the cluster
+       |  |     |--config       # Host level config
+       |  |     |  `--marker
```
Comment (TrimBiggs) [inline docs/etcdStructure.md]: Also have this `DefaultEndpointToHostAction` under config here.

File: docs/etcdStructure.md [hunk 5]
```diff
@@ -9,148 +9,233 @@
...
-	            |--ip_addr_v6 # the IP address BIRD6 listens on
-	            |--as_num     # the AS number for this host
-	            |--peer_v4    # Host specific IPv4 BGP peers
-	            |  `--<BGP peer IPv4 address>  # JSON BGP peer configuration (see below)
-	            `--peer_v6  # Host specific IPv6 BGP peers
-	               `--<BGP peer IPv6 address>  # JSON BGP peer configuration (see below)
+    
+    +--calico  # root namespace
+       |
+       |--v1
```
Comment (TrimBiggs) [inline docs/etcdStructure.md]: A `Ready` flag here too:
`/calico/v1/Ready`

**Comment by TrimBiggs** [comment]
Should the values for `/calico/felix` be in here also?
```
/calico/felix/v1/host/d41617dcb07d/last_reported_status
/calico/felix/v1/host/d41617dcb07d/status
```

**Comment by TrimBiggs** [comment]
Layout looks good.  Made some comments about some additional contents to add.

File: docs/etcdStructure.md (hunk 3 above)
Comment (robbrockbank) [inline docs/etcdStructure.md]: Ah that's what you meant!  Actually yes I think this should just be removed.   I think the bird_ip should probably be removed from the code too.  Need to check if calico felix is still using it.

File: docs/etcdStructure.md (hunk 5 above)
Comment (robbrockbank) [inline docs/etcdStructure.md]: Not in the main calico docs, i'm omitting for now.

File: docs/etcdStructure.md (hunk 4 above)
Comment (robbrockbank) [inline docs/etcdStructure.md]: I have added a line above that points to the main read the docs which describes all of the options, including this one.

**Comment by robbrockbank** [comment]
The last_reported_status and status are not documented at the moment, so I'm omitting.

**Comment by robbrockbank** [comment]
Errors fixed, and some more details added.  Merging.

**Pull request merged**